core/state/trie_prefetcher.go

```diff
@@ -561,17 +548,17 @@ func (to *trieOrchestrator) processTasks() {
 
 			// Enqueue task for processing (may spawn new goroutine
 			// if not at [maxConcurrency])
-			if !to.sf.p.workers.Execute(to.ctx, f) {
-				to.restoreOutstandingRequests(len(tasks[i:]))
-				return
-			}
+			//
+			// If workers are stopped before calling [Execute], this function may
+			// block forever.
+			to.sf.p.workers.Execute(f)
```

Comment:
If workers are stopped before calling `Execute` this will panic when trying to pass `f` to a closed channel, no?

Reply:
Yes, you are right. I'll update this comment.